BUILDING.md

```diff
@@ -49,6 +49,31 @@ cmake --build build --config Release
 
 If you are using vcpkg as your C++ package manager, you might need to add `-DCMAKE_TOOLCHAIN_FILE=<path-to-your-vcpkd-root>/scripts/buildsystems/vcpkg.cmake` to the CMake command in the configuration step.
 
+### Unit Testing
+
+For configuring with testing enabled (only MSVC is enabled for now):
+
+```sh
+mkdir build
+cd build
+cmake .. --preset=test-msvc
+cmake --build test
+ctest -C debug test
+```
+
+### vcpkg (for testing)
```

Comment:
I thought we came to agreement that testing shouldn't require VCPKG?

Reply:
We don't, this is just one documented way of building.

Reply:
> We don't, this is just one documented way of building.

Excellent.  Maybe that should be reflected in the heading, along the lines of "### Using VCPKG for testing" or something similar, and corresponding instructions for when not using VCPKG.

Reply:
I believe we can actually refer back to the section which talks about acquiring GSL here.  The recommendation is exactly the same.  I will enumerate some of the commonly used package names for doctest though.